### PR TITLE
Add Member Toggle for Admins

### LINKED ISSUE #92: Create way for eboard/other admin roles to switch into a "regular user mode"
It would be nice if there was a toggle switch at the top of the screen that would allow for eboard members to go around the website in "regular user mode", or some mode that does not show hidden files at all. This way, an eboard member could display gallery on a public screen without worrying about displaying hidden content.

### PR DESCRIPTION
**Pull request by jabbate19**
"I want a way to go on Gallery on a projector, scrolling through random files and folders and not risk showing something hidden. Basically a way to act as a member without special perms"
- @mbreidenbach (Kinda, to some extent. I think this was his words)

Fixes #92

### CODE REVIEW

File: gallery/__init__.py [hunk 1]
```diff
@@ -959,17 +970,21 @@ def render_dir(dir_id: int, auth_dict: Optional[Dict[str, Any]] = None):
...
+                           lockdown=gallery_lockdown,
+                           member_lock=member_lock)
 
 
 @app.route("/view/file/<int:file_id>")
 @auth.oidc_auth('default')
 @gallery_auth
 def render_file(file_id: int, auth_dict: Optional[Dict[str, Any]] = None):
+    member_lock = session['userinfo'].get('member-lock')
+    print(session)
```
Comment (mxmeinhold) [inline gallery/__init__.py:982]: debug print?

File: gallery/__init__.py [hunk 2]
```diff
@@ -1040,11 +1059,24 @@ def get_member_list(auth_dict: Optional[Dict[str, Any]] = None):
...
 
 
+@app.route("/member", methods = ["POST"])
+@auth.oidc_auth('default')
+@gallery_auth
+def member_mode(auth_dict: Optional[Dict[str, Any]] = None):
+    if session['userinfo'].get('member-lock'):
+        session['userinfo']['member-lock'] = not session['userinfo']['member-lock']
+    else:
+        session['userinfo']['member-lock'] = True
```
Comment (mxmeinhold) [inline gallery/__init__.py:1069]:
```suggestion
    session['userinfo'].get('member-lock') = not session['userinfo']['member-lock']
```
this does the same thing the if did

File: gallery/__init__.py [hunk 3]
```diff
@@ -128,6 +128,8 @@ def index():
 @auth.oidc_auth('default')
 @gallery_auth
 def view_upload(auth_dict: Optional[Dict[str, Any]] = None):
+    member_lock = session['userinfo'].get('member-lock')
+
```
Comment (mxmeinhold) [inline gallery/__init__.py:132]: all the places where you're doing this and not using member_lock anywhere but the render_template, why not skip the variable and just do `session....get...` in the render_template args?

File: gallery/templates/view_dir.html [hunk 4]
```diff
@@ -35,7 +35,7 @@
 {% if children|length >= 1 %}
     <div class="row" id="child_list">
     {% for child_type, child in children %}
-        {% if not child.hidden or (auth_dict['is_eboard'] or auth_dict['is_rtp'])  or auth_dict['is_alumni'] %}
+        {% if not child.hidden or (auth_dict['is_eboard'] or auth_dict['is_rtp'] or auth_dict['is_alumni']) and not member_lock %}
```
Comment (mxmeinhold) [inline gallery/templates/view_dir.html:38]:
```suggestion
        {% if not child.hidden or (auth_dict['is_eboard'] or auth_dict['is_rtp'] or auth_dict['is_alumni']) and not session['userinfo'].get('member-lock') %}
```
does this work? can't remember if session will be defined in the context of jinja

File: gallery/templates/nav.html [hunk 5]
```diff
@@ -37,6 +37,11 @@
                                     {% else %}
                                     <li><a href="#" onclick="galleryLockdown(event);"><span class="glyphicon glyphicon-lock"></span> Lockdown</a></li>
                                 {% endif %}
+                                {% if member_lock %}
+                                    <li><a href="#" onclick="memberModeToggle(event);"><span class="glyphicon glyphicon-star"></span> Admin Mode</a></li>
+                                    {% else %}
+                                    <li><a href="#" onclick="memberModeToggle(event);"><span class="glyphicon glyphicon-user"></span> Member Mode</a></li>
+                                {% endif %}
```
Comment (mxmeinhold) [inline gallery/templates/nav.html:44]: nit: say `Switch To {} Mode` for clarity?

File: gallery/__init__.py [hunk 6]
```diff
@@ -1040,11 +1059,24 @@ def get_member_list(auth_dict: Optional[Dict[str, Any]] = None):
     return jsonify(ldap.get_members())
 
 
+@app.route("/member", methods = ["POST"])
```
Comment (mxmeinhold) [inline gallery/__init__.py:1062]: could we call this like... `member_mode_toggle`?

File: gallery/__init__.py (hunk 2 above)
Comment (jabbate19) [reply gallery/__init__.py:1069]: Not if it doesn't exist, this isn't pre-added to user info so we need to init it to the dict

Comment (mxmeinhold) [reply gallery/__init__.py:1069]: oh, maybe this to do the default?
```suggestion
    session['userinfo']['member-lock'] = not session['userinfo'].get('member-lock', False)
```

File: gallery/__init__.py [hunk 7]
```diff
@@ -970,21 +959,19 @@ def render_dir(dir_id: int, auth_dict: Optional[Dict[str, Any]] = None):
...
-    member_lock = session['userinfo'].get('member-lock')
-    print(session)
-
     file_model = File.query.filter(File.id == file_id).first()
     if file_model is None:
         abort(404)
-    if file_model.hidden and ((not auth_dict['is_eboard'] and not auth_dict['is_rtp'] and not auth_dict['is_alumni']) or member_lock):
+    if file_model.hidden and ((not auth_dict['is_eboard']
+                               and not auth_dict['is_rtp']
+                               and not auth_dict['is_alumni']) or session['userinfo'].get('member-lock')):
```
Comment (mxmeinhold) [inline gallery/__init__.py:974]: should this be `.get('member-lock', False)` for the default?

File: gallery/templates/nav.html [hunk 8]
```diff
@@ -37,10 +37,10 @@
                                     {% else %}
                                     <li><a href="#" onclick="galleryLockdown(event);"><span class="glyphicon glyphicon-lock"></span> Lockdown</a></li>
                                 {% endif %}
-                                {% if member_lock %}
-                                    <li><a href="#" onclick="memberModeToggle(event);"><span class="glyphicon glyphicon-star"></span> Admin Mode</a></li>
+                                {% if session['userinfo']['member-lock'] %}
```
Comment (mxmeinhold) [inline gallery/templates/nav.html:40]: does this need to be `.get` in case it's not in the userinfo dict?

File: gallery/templates/view_dir.html [hunk 9]
```diff
@@ -35,7 +35,7 @@
 {% if children|length >= 1 %}
     <div class="row" id="child_list">
     {% for child_type, child in children %}
-        {% if not child.hidden or (auth_dict['is_eboard'] or auth_dict['is_rtp'] or auth_dict['is_alumni']) and not member_lock %}
+        {% if not child.hidden or (auth_dict['is_eboard'] or auth_dict['is_rtp'] or auth_dict['is_alumni']) and not session['userinfo']['member-lock'] %}
```
Comment (mxmeinhold) [inline gallery/templates/view_dir.html:38]: ditto, `.get`

File: gallery/__init__.py (hunk 7 above)
Comment (jabbate19) [reply gallery/__init__.py:974]: False and None to work the same in this context

**Review by mxmeinhold — APPROVED** [review]

**Pull request merged**